workbench/src/renderer/app.jsx

```diff
@@ -36,446 +36,369 @@ const { ipcRenderer } = window.Workbench.electron;
       // Show changelog if this is a new version,
       // but if it's the first run ever, wait until after download modal closes.
-      showChangelog: this.props.isNewVersion && !this.props.isFirstRun,
-    });
-    await i18n.changeLanguage(window.Workbench.LANGUAGE);
-    ipcRenderer.on('download-status', (downloadedNofN) => {
-      this.setState({
-        downloadedNofN: downloadedNofN,
+      setShowChangelog(props.isNewVersion && !props.isFirstRun);
+      await i18n.changeLanguage(window.Workbench.LANGUAGE);
+      ipcRenderer.on('download-status', (downloadedNofN) => {
+        setDownloadedNofN(downloadedNofN)
       });
-    });
-  }
+    }
+    componentDidMount();
 
-  componentWillUnmount() {
-    ipcRenderer.removeAllListeners('download-status');
-  }
+    // Cleanup function to execute on component unmount
+    return () => { ipcRenderer.removeAllListeners('download-status'); }
+  }, []);
 
-  /**
-   * Change the tab that is currently visible.
-   * @param {string} key - the value of one of the Nav.Link eventKey.
-   */
-  switchTabs(key) {
-    this.setState(
-      { activeTab: key }
-    );
-  }
+  // When investList changes, re-filter the recent jobs according to the new list
+  useEffect(() => {
+    updateRecentJobs();
+  }, [investList]);
 
-  toggleDownloadModal(shouldShow) {
-    this.setState({
-      showDownloadModal: shouldShow,
-    });
+  function toggleDownloadModal(shouldShow) {
+    setShowDownloadModal(shouldShow);
     // After close, show changelog if new version and app has just launched
     // (i.e., show changelog only once, after the first time the download modal closes).
-    if (!shouldShow && this.props.isNewVersion && !this.state.changelogDismissed) {
-      this.setState({
-        showChangelog: true,
-      });
+    if (!shouldShow && props.isNewVersion && !changelogDismissed) {
+      setShowChangelog(true);
     }
   }
 
-  closeChangelogModal() {
-    this.setState({
-      showChangelog: false,
-      changelogDismissed: true,
-    });
-  }
-
-  togglePluginModal(show) {
-    this.setState({
-      showPluginModal: show
-    });
-  }
-
-  toggleMetadataModal(show) {
-    this.setState({
-      showMetadataModal: show
-    });
-  }
-
-  toggleSettingsModal(show) {
-    this.setState({
-      showSettingsModal: show
-    });
+  function closeChangelogModal() {
+    setShowChangelog(false);
+    setChangelogDismissed(true);
   }
 
   /**
    * Push data for a new InvestTab component to an array.
    * @param {InvestJob} job - as constructed by new InvestJob()
    */
-  openInvestModel(job) {
+  function openInvestModel(job) {
     const tabID = window.crypto.getRandomValues(
       new Uint32Array(1)
     ).toString();
-    const { openJobs, openTabIDs } = this.state;
-    openTabIDs.push(tabID);
-    openJobs[tabID] = job;
-    this.setState({
-      openTabIDs: openTabIDs,
-      openJobs: openJobs,
-    }, () => this.switchTabs(tabID));
+    const newOpenJobs = new Map(openJobs);
```

Comment:
I also took the opportunity to combine the `openTabIDs` and `openJobs` state variables into a single `Map` which serves the purpose of both (tracking both the order of open tabs and their corresponding jobs).